features/newts/src/main/java/org/opennms/netmgt/measurements/impl/NewtsFetchStrategy.java

```diff
@@ -411,4 +421,30 @@ protected void setSampleRepository(SampleRepository sampleRepository) {
     protected void setContext(Context context) {
         m_context = context;
     }
+
+    private OnmsNode getNode(final OnmsResource resource, final Source source) {
+        OnmsNode node = null;
+        try {
+            node = ResourceTypeUtils.getNodeFromResourceRoot(resource);
+        } catch (final ObjectRetrievalFailureException e) {
+        }
+        if (node == null) {
+            final String resourceId = source.getResourceId().split("\\.")[0];
```

Comment:
Can we use the `ResourceId` class to parse this instead of tokenizing it ourself here? Something like `ResourceId.fromString(resourceId).getParent()`.

Reply:
Ah yeah, that makes sense.